braindecode/datautil/serialization.py

```diff
@@ -105,14 +105,40 @@ def _load_signals_and_description(path, preload, raws, ids_to_load=None):
             [int(os.path.split(f)[-1].split('-')[0]) for f in file_names])
     for i in ids_to_load:
         fif_file = os.path.join(path, file_name.format(i))
-        all_signals.append(_load_signals(fif_file, preload, raws))
+        all_signals.append(_load_signals(fif_file, preload))
     description_df = description_df.iloc[ids_to_load]
     return all_signals, description_df
 
 
-def _load_signals(fif_file, preload, raws):
-    if raws:
+def _load_signals(fif_file, preload):
+    if fif_file.endswith('-raw.fif'):
         signals = mne.io.read_raw_fif(fif_file, preload=preload)
-    else:
+    elif fif_file.endswith('-epo.fif'):
         signals = mne.read_epochs(fif_file, preload=preload)
+    else:
+        raise ValueError('fif_file must end with raw.fif or epo.fif.')
     return signals
+
+
+def _check_save_dir_empty(save_dir):
+    """Make sure a BaseConcatDataset can be saved under a given directory.
+
+    Parameters
+    ----------
+    save_dir : str
+        Directory under which ``concat_ds`` will be saved.
+
+    Raises
+    -------
+    FileExistsError
+        If ``save_dir`` is not a valid directory for saving, i.e., the
+        description.json and/or target_name.json files already exist.
```

Comment:
Our logic changed a bit here. No file is allowed to exist in `save_dir`. Subdirectories are fine.